### PR TITLE
Support packs in secrets

### PR DESCRIPTION
**Pull request by yardensade**
## Status
Ready

fixes: https://github.com/demisto/etc/issues/19890

### CODE REVIEW

File: Tests/scripts/hook_validations/secrets.py [hunk 1]
```diff
@@ -259,20 +273,31 @@ def calculate_shannon_entropy(data):
...
 
 
-def get_white_list():
-    with io.open('./Tests/secrets_white_list.json', mode="r", encoding="utf-8") as secrets_white_list_file:
-        final_white_list = []
-        ioc_white_list = []
-        files_while_list = []
+def get_white_listed_items(is_pack, pack_name):
+    if is_pack:
+        packs_whitelist_path = PACKS_PATH + pack_name + PACKS_WHITELIST_RELATIVE_PATH
```
Comment (yaakovi) [inline Tests/scripts/hook_validations/secrets.py]: use `os.path.join` and not hard-coded `/`

File: Tests/scripts/hook_validations/secrets.py [hunk 2]
```diff
@@ -120,39 +126,41 @@ def search_potential_secrets(secrets_file_paths):
...
+        if is_pack:
+            file_contents = remove_white_list_regexs(file_contents, secrets_white_list)
+            secrets_white_list = set()
         # Validate if it is integration documentation file
-        integration_readme = re.match(pattern=INTEGRATION_README_REGEX,
-                                      string=file_path,
-                                      flags=re.IGNORECASE)
-        # if py/js file, search for yml in order to retrieve temp white list
-        if file_extension in {'.py', '.js'} or integration_readme:
+        integration_readme = re.match(pattern=INTEGRATION_README_REGEX, string=file_path, flags=re.IGNORECASE)
```
Comment (yaakovi) [inline Tests/scripts/hook_validations/secrets.py]: what about `PACK_README_REGEX`?
use `check_type` function

File: Tests/scripts/hook_validations/secrets.py [hunk 3]
```diff
@@ -365,6 +401,15 @@ def ignore_base64(file_contents):
     return file_contents
 
 
+def is_file_path_in_pack(file_path):
+    return bool(re.findall(r'^(?:./)?{}/'.format(PACKS_DIR), file_path))
```
[the remark this answers is not in the document]
Comment (yardensade) [reply Tests/scripts/hook_validations/secrets.py]: We don't and this didn't feel like regex we will need outside this scope, maybe add this as func in utils? Not sure.

File: Tests/scripts/hook_validations/secrets.py [hunk 4]
```diff
@@ -16,9 +16,15 @@
 ENTROPY_THRESHOLD = 4.0
 ACCEPTED_FILE_STATUSES = ['m', 'a']
 SKIPPED_FILES = {'secrets_white_list', 'id_set.json', 'conf.json', 'Pipfile', 'secrets-ignore'}
-TEXT_FILE_TYPES = {'.yml', '.py', '.json', '.md', '.txt', '.sh', '.ini', '.eml', '', '.csv', '.js', '.pdf', '.html'}
+TEXT_FILE_TYPES = {'.yml', '.py', '.json', '.md', '.txt', '.sh', '.ini', '.eml', '', '.csv', '.js', '.pdf', '.html',
```
[the remark this answers is not in the document]
Comment (yardensade) [reply Tests/scripts/hook_validations/secrets.py]: There are some files that have no extension giving back empty string to the `os.path.splitext(file_path)[1]`, itay added pipfile to files being ignored completely so we are not looking to scan those.

**Comment by yardensade** [comment]
> very nice.
> 
> lets open an issue for improving performance in the script.

I already stated them in https://github.com/demisto/etc/issues/19890, after we are done I will open concentrated left overs in a single issue.

File: Tests/scripts/hook_validations/secrets.py (hunk 2 above)
Comment (yardensade) [reply Tests/scripts/hook_validations/secrets.py]: I'm a bit conflicted about this, the checked_type we use right now is not for file types more like Demisto types so it's not a good solution for me.
Doing my own check_type would require me to change my file extension set to regexes for each type, is that what we are looking for? Don't see any other use for this data atm. (or to send both file_path & file extension into check type)

File: Tests/scripts/hook_validations/secrets.py (hunk 1 above)
Comment (yardensade) [reply Tests/scripts/hook_validations/secrets.py]: Sorry you had to see this 😞

File: Tests/scripts/hook_validations/secrets.py (hunk 2 above)
Comment (yardensade) [reply Tests/scripts/hook_validations/secrets.py]: Went for hybrid solution

File: Tests/scripts/constants.py [hunk 5]
```diff
@@ -250,6 +251,11 @@
 
 SCRIPTS_REGEX_LIST = [SCRIPT_YML_REGEX, SCRIPT_PY_REGEX, SCRIPT_JS_REGEX, SCRIPT_PS_REGEX]
 
+# All files that have related yml file
+REQUIRED_YML_FILE_TYPES = [SCRIPT_PY_REGEX, INTEGRATION_PY_REGEX, PACKS_INTEGRATION_PY_REGEX, PACKS_SCRIPT_PY_REGEX,
+                           SCRIPT_JS_REGEX, INTEGRATION_JS_REGEX, PACKS_SCRIPT_JS_REGEX, PACKS_INTEGRATION_JS_REGEX,
+                           PACKS_README_REGEX, INTEGRATION_README_REGEX]
```
Comment (yaakovi) [inline Tests/scripts/constants.py]: what about powershell? 😉

**Comment by yardensade** [comment]
I will leave it for when we start developing infra lol, didn't wanna mix
too much in the end ;)

File: Tests/scripts/constants.py (hunk 5 above)
Comment (yardensade) [reply Tests/scripts/constants.py]: I will leave it for when we start developing infra lol, didn't wanna mix
too much in the end ;)

File: Tests/scripts/hook_validations/secrets.py [hunk 6]
```diff
@@ -209,6 +205,21 @@ def create_temp_white_list(file_contents):
...
+    yml_file_contents = ''
+    # if script or readme file, search for yml in order to retrieve temp white list
+    if is_allowed_types(file_path):
+        yml_file_contents = retrieve_related_yml(os.path.dirname(file_path))
+    return yml_file_contents
+
+
+def is_allowed_types(file_path):
+    # Validate if it is integration documentation file or supported file extension
+    if any(re.match(regex, file_path, re.IGNORECASE) for regex in REQUIRED_YML_FILE_TYPES):
```
Comment (yaakovi) [inline Tests/scripts/hook_validations/secrets.py]: again, `test_utils:checked_type`

Comment (yardensade) [reply Tests/scripts/hook_validations/secrets.py]: image

Comment (yardensade) [reply Tests/scripts/hook_validations/secrets.py]: Done, but I like my version better

**Pull request merged**